### PR TITLE
Update django template editor to match web editor API

### LINKED ISSUE #5326: Update django IntelliSense for web editor breaking changes
Some of the internal APIs that we are using are changing, we need to update our workload at the same time as the web editor team.

### PR DESCRIPTION
**Pull request by huguesv**
Fix #5326

### CODE REVIEW

File: Python/Product/Django/Intellisense/TemplateCommandFactory.cs [hunk 1]
```diff
@@ -14,17 +14,14 @@
 // See the Apache Version 2.0 License for specific language governing
 // permissions and limitations under the License.
 
+#if !DEV16_OR_LATER
```
Comment (ToddGrun) [inline Python/Product/Django/Intellisense/TemplateCommandFactory.cs]: If you're not worrying about Dev15 builds, can you just get rid of this file?

Comment (huguesv) [reply Python/Product/Django/Intellisense/TemplateCommandFactory.cs]: Agreed, I'll remove the django dev15 only code.

File: Python/Product/Django/DjangoEditorFactory.cs [hunk 2]
```diff
@@ -16,11 +16,7 @@
 
 using System;
 using System.Runtime.InteropServices;
-#if DEV16_OR_LATER
```
Comment (MinKPark) [inline Python/Product/Django/DjangoEditorFactory.cs]: does it mean we don't support less version than Dev16 or the package is just work for both? I wonder.

Comment (huguesv) [reply Python/Product/Django/DjangoEditorFactory.cs]: The builds are specific to a version of VS.

We have the 15.9 branch which targets VS 2017.

The master branch only targets VS 2019.

Until VS 2019 was GA, it was useful to allow build/run master branch on VS 2017, but now that VS 2019 is officially out, we don't need to keep maintaining support for VS 2017 in master branch.

**Pull request merged**